packages/context/src/lib/controllers/context-provider.ts

```diff
@@ -82,7 +82,12 @@ export class ContextProvider<
       this.context = contextOrOptions as T;
     }
     this.attachListeners();
-    this.host.addController(this);
+    // We may be constructed with a plain DOM element to provide context to a
+    // whole DOM tree of elements, in this case the host DOM element doesn't
+    // need know about this controller.
+    if ((host as ReactiveControllerHost).addController !== undefined) {
+      this.host.addController(this);
+    }
```

Comment:
Optional chaining is shorter here.
```suggestion
    this.host.addController?.(this);
```

The comment should be moved to the jsdoc above the `ContextProvider` and more user facing.

Something like..
> The controller may also be attached to any HTML element in which case it's up to the user to call `hostConnected()` when attached to the DOM. This is done automatically for any custom elements implementing `ReactiveControllerHost`.